lib/adapters/file/write.js

```diff
@@ -0,0 +1,135 @@
+var _ = require('underscore');
+var Juttle = require('../../runtime/index').Juttle;
+var Promise = require('bluebird');
+var fs = Promise.promisifyAll(require('fs'));
+
+var Write = Juttle.proc.sink.extend({
+    procName: 'write-file',
+    initialize: function(options, params) {
+        var allowed_options = ['file', 'limit', 'maxFilesize'];
+        var unknown = _.difference(_.keys(options), allowed_options);
+        if (unknown.length > 0) {
+            throw this.compile_error('RT-UNKNOWN-OPTION-ERROR', {
+                proc: 'write file',
+                option: unknown[0]
+            });
+        }
+
+        if (!_.has(options, 'file')) {
+            throw this.compile_error('RT-MISSING-OPTION-ERROR', {
+                proc: 'write file',
+                option: "file"
+            });
+        }
+
+        this.limit = options.limit || 100000;
+        if (!_.isNumber(this.limit) || this.limit < 0 ) {
+            throw this.compile_error('RT-OPTION-SHOULD-BE-POSITIVE-INTEGER', {
+                option: "limit"
+            });
+        }
+
+        // 200MB limit on file
+        this.maxFilesize = options.maxFilesize || 200*1024*1024;
+        if (!_.isNumber(this.maxFilesize) || this.maxFilesize < 0 ) {
+            throw this.compile_error('RT-OPTION-SHOULD-BE-POSITIVE-INTEGER', {
+                option: "maxFilesize"
+            });
+        }
+
+        this.filename = options.file;
+        this.queue = [];
+    },
+
+    start: function() {
+        this.flush();
+    },
+
+    process: function(points) {
+        this.logger.debug('process', points);
+        this.queue = this.queue.concat(points);
+    },
+
+    flush: function(final) {
+        var self = this;
+
+        if (this.isdone) {
+            return;
+        }
+
+        if (final) {
+            this.isdone = true;
+        }
+
+        if (this.queue.length !== 0) {
+            var points = [];
+
+            fs.statAsync(this.filename)
+            .then(function(stats) {
+                if (stats.size > self.maxFilesize) {
+                    throw Error('Exceeded max filesize of ' + self.maxFilesize + ' bytes');
```

Comment:
This seems like it should trigger the `RT-PROC-LIMIT-EXCEEDED` error but from my read of the code it's going to show up as an internal error.